.travis.yml

```diff
@@ -2,31 +2,48 @@ dist: xenial
 sudo: false
 language: scala
 
-scala:
-  - 2.12.8
-  - 2.11.12
+scala_version_211: &scala_version_211 2.11.12
+scala_version_212: &scala_version_212 2.12.8
+scala_version_213: &scala_version_213 2.13.0-M5
+scala_version_docs: &scala_version_docs 2.12.8
+  
+java11: &java_11 openjdk11
+java12: &java_12 openjdk12
 
-jdk:
-  - openjdk11
-  - openjdk12
+stages:
+  - name: test
+  - name: publish
+    if: (branch = master AND type = push)
 
-before_install:
-  - export PATH=${PATH}:./vendor/bundle
+jobs:
+  include:
+    - &tests
+      env: TEST="test"
+      script:
+        - sbt ++$TRAVIS_SCALA_VERSION test
+        - sbt ++$TRAVIS_SCALA_VERSION mimaReportBinaryIssues
+      scala: *scala_version_211
+      jdk: *java_11
+    - <<: *tests
+      scala: *scala_version_212
+    - <<: *tests
+      scala: *scala_version_213
+    - <<: *tests
+      scala: *scala_version_212
+      jdk: *java_12
 
-install:
-  - rvm use 2.6.0 --install --fuzzy
-  - gem update --system
-  - gem install sass
-  - gem install jekyll -v 3.2.1
+    - env: TEST="docs"
+      scala: *scala_version_docs
+      script:
+        - sbt ++$TRAVIS_SCALA_VERSION docs/makeSite
 
-script:
-  - sbt ++$TRAVIS_SCALA_VERSION test
-  - sbt ++$TRAVIS_SCALA_VERSION mimaReportBinaryIssues
-  - sbt ++$TRAVIS_SCALA_VERSION docs/makeSite
-
-after_success:
-  - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_JDK_VERSION == "openjdk11" && test $TRAVIS_BRANCH == "master" && test $TRAVIS_REPO_SLUG == "http4s/http4s-jdk-http-client" && sbt ++$TRAVIS_SCALA_VERSION publish
-  - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_JDK_VERSION == "openjdk11" && test $TRAVIS_BRANCH == "master" && test $TRAVIS_REPO_SLUG == "http4s/http4s-jdk-http-client" && test $TRAVIS_SCALA_VERSION == "2.12.8" && sbt docs/ghpagesPushSite
+    - &publish
+      stage: publish
+      env: TEST="publish"
+      jdk: *java_11
+      scala: *scala_version_docs
+      script:
+        - sbt ++$TRAVIS_SCALA_VERSION ;+publishSigned ;docs/makeSite ;docs/ghpagesPushSite
```

Comment:
Folded this back into one step, or else we'll have to recompile the entirety of the code again.

None of this is using the sbt-release architecture we already have set up in the build.  That's an unresolved tension here.